test/src/main/java/org/apache/accumulo/test/metrics/MetricsIT.java

```diff
@@ -232,6 +258,186 @@ public void confirmMetricsPublished() throws Exception {
+        // total = configured size
+        Wait.waitFor(() -> {
+          for (var line : sink.getLines()) {
+            TestStatsDSink.Metric metric = TestStatsDSink.parseStatsDMetric(line);
+            // if the metric is one of the fate executor metrics...
+            if (metric.getName().equals(FATE_OPS_THREADS_INACTIVE.getName())
+                || metric.getName().equals(FATE_OPS_THREADS_TOTAL.getName())) {
+              var tags = metric.getTags();
+              var instanceType = FateInstanceType
+                  .valueOf(tags.get(FateExecutorMetrics.INSTANCE_TYPE_TAG_KEY).toUpperCase());
+              var opsAssigned = tags.get(FateExecutorMetrics.OPS_ASSIGNED_TAG_KEY);
+
+              verifyFateMetricTags(opsAssigned, instanceType);
+
+              if (metric.getName().equals(FATE_OPS_THREADS_TOTAL.getName())) {
+                assertEquals(numFateThreadsPool1, Integer.parseInt(metric.getValue()));
+                if (instanceType == FateInstanceType.USER) {
+                  sawTotalThreadsUserMetric.set(true);
+                } else if (instanceType == FateInstanceType.META) {
+                  sawTotalThreadsMetaMetric.set(true);
+                }
+              } else if (metric.getName().equals(FATE_OPS_THREADS_INACTIVE.getName())) {
+                if (instanceType == FateInstanceType.USER
+                    && (Integer.parseInt(metric.getValue()) == numFateThreadsPool1 - 1)) {
+                  sawExpectedInactiveUserThreads.set(true);
+                } else if (instanceType == FateInstanceType.META
+                    && (Integer.parseInt(metric.getValue()) == numFateThreadsPool1 - 1)) {
+                  sawExpectedInactiveMetaThreads.set(true);
+                }
+              }
+            }
+          }
+
+          log.debug("sawExpectedInactiveUserThreads: " + sawExpectedInactiveUserThreads.get());
+          log.debug("sawExpectedInactiveMetaThreads: " + sawExpectedInactiveMetaThreads.get());
+          log.debug("sawTotalThreadsUserMetric: " + sawTotalThreadsUserMetric.get());
+          log.debug("sawTotalThreadsMetaMetric: " + sawTotalThreadsMetaMetric.get());
+          return sawExpectedInactiveUserThreads.get() && sawExpectedInactiveMetaThreads.get()
+              && sawTotalThreadsUserMetric.get() && sawTotalThreadsMetaMetric.get();
+        });
+
+        // Now change the config from:
+        // {<all fate ops>: numFateThreadsPool1}
+        // ->
+        // {<all fate ops except split>: numFateThreadsPool2,
+        // <split operation>: numFateThreadsPool3}
+        changeFateConfig(client, FateInstanceType.USER);
+        changeFateConfig(client, FateInstanceType.META);
+
+        // Wait for config changes to get picked up by FATE. Can do so by waiting for metrics to no
+        // longer include metrics of the first pool
+        Wait.waitFor(() -> {
+          boolean metricExistsForPool1 = false;
+          for (var line : sink.getLines()) {
```

Comment:
The way this is looking for the absence of something may not be correct because metrics arrive periodically.  So when something is not seen it could be because it has not arrived yet.  To know for sure that something is absent may need to wait for some multiple of the statsd period and then look at the metrics that have built up.

Reply:
Maybe could combine the test code for looking for new stuff and absence of old stuff.  Like if we see new stuff and no old stuff two or three times in a row, then its probably good.